### PR TITLE
Fix: restrict simulator embed/show to HTML and reject non-HTML formats (#6352)

### LINKED ISSUE #6352: ActionView::MissingTemplate: Missing template simulator/embed, application/embed with {:locale=>[:en], :formats=>[:png], :variants=>[], :handlers=>[:raw, :erb, :html, :builder, :ruby, :coffee, :jbuilder]}. (ActionView::MissingTemplate)
Sentry Issue: [CIRCUITVERSE-CORE-TF](https://circuitverse.sentry.io/issues/77799984/)

```
ActionView::MissingTemplate: Missing template simulator/embed, application/embed with {:locale=>[:en], :formats=>[:png], :variants=>[], :handlers=>[:raw, :erb, :html, :builder, :ruby, :coffee, :jbuilder]}. (ActionView::MissingTemplate)

Searched in:
  * "/home/ubuntu/main/CircuitVerse/app/views"
  * "/home/ubuntu/.rvm/gems/ruby-3.3.0/gems/maintenance_tasks-2.12.0/app/views"
  * "/home/ubuntu/.rvm/gems/ruby-3.3.0/bundler/gems/simple_discussion-3bd69f1e58f7/app/views"
  * "/home/ubuntu/.rvm/gems/ruby-3.3.0/gems/rails_admin-3.1.3/app/views"
  * "/home/ubuntu/.rvm/gems/ruby-3.3.0/gems/turbo-rails-1.5.0/app/views"
  * "/home/ubuntu/.rvm/gems/ruby-3.3.0/gems/kaminari-core-1.2.2/app/views"
  * "/home/ubuntu/.rvm/gems/ruby-3.3.0/gems/commontator-7.0.0/app/views"
  * "/home/ubuntu/.rvm/gems/ruby-3.3.0/gems/devise-4.9.2/app/views"
  * "/home/ubuntu/.rvm/gems/ruby-3.3.0/gems/view_component-3.9.0/app/views"
  * "/home/ubuntu/.rvm/gems/ruby-3.3.0/gems/mailkick-0.4.3/app/views"
  * "/home/ubuntu/.rvm/gems/ruby-3.3.0/gems/actiontext-7.0.8.7/app/views"
  * "/home/ubuntu/.rvm/gems/ruby-3.3.0/gems/actionmailbox-7.0.8.7/app/views"

  app/controllers/simulator_controller.rb:46:in `embed'
    render "embed"
  app/controllers/application_controller.rb:32:in `switch_locale'
    I18n.with_locale(locale, &)
...
(120 additional frame(s) were not displayed)
```
This issue was automatically created by Sentry via [New issue](https://sentry.io/organizations/circuitverse/alerts/rules/circuitverse-core/11296/)

### PR DESCRIPTION
**Pull request by naman79820**
Fixes #6352 

#### Describe the changes you have made in this PR -

This PR fixes the ActionView::MissingTemplate error occurring when requests were made to /simulator/embed with non-HTML formats (such as .png, ?format=png, or Accept: image/png).

Changes include:

Updated SimulatorController (show and embed actions) to explicitly handle only HTML responses using respond_to blocks.

Return HTTP 406 (Not Acceptable) for unsupported formats.

Updated routes.rb to disable automatic format matching (format: false) for simulator routes to prevent .png extensions from being routed.

Verified with curl tests:

/simulator/embed/:id → 200

/simulator/embed/:id.png → 404

/simulator/embed/:id?format=png → 406

Accept: image/png → 406

Removed internal server error (500) and ensured consistent behavior for invalid formats.

**Did you use AI assistance (ChatGPT, Claude, Copilot, etc.) to write any part of this code?**
- [ ] No, I wrote all the code myself
- [x] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [x] I have reviewed every single line of the AI-generated code
- [x] I can explain the purpose and logic of each function/component I added
- [x] I have tested edge cases and understand how the code handles them
- [x] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**
The issue occurred because Rails attempted to render a non-existent view template (simulator/embed.png) when requests with unsupported formats were received.
To fix this, the show and embed actions in SimulatorController were modified to respond only with HTML, returning 406 Not Acceptable for other formats.
Additionally, the routes.rb file was updated to include format: false to prevent .png and similar extensions from being routed to the controller.
This ensures that the app gracefully handles invalid format requests without throwing internal server errors.

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

Note: Please check **Allow edits from maintainers** if you would like us to assist in the PR.

## Summary by CodeRabbit

* **Refactor**
  * Standardized response handling for simulator "show" and "embed" endpoints so they consistently render the intended HTML view.
  * Updated routing to remove format suffixes from simulator URLs, ensuring cleaner links and making non-HTML format requests receive a rejection (not served).

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by naman79820** [comment]
I'll try to implement coderabbit suggestions by this evening 😄

**Comment by naman79820** [comment]
@tachyons sir, would appreciate your review whenever you get a chance :))

File: config/routes.rb [hunk 1]
```diff
@@ -143,7 +143,7 @@
 
   # simulator
   scope "/simulator" do
-    get "/:id", to: "simulator#show", as: "simulator"
+    get "/:id", to: "simulator#show", as: "simulator" , format: false
```
Comment (tachyons) [inline config/routes.rb:146]: Do you still need changes on controller after making this change ?

**Comment by naman79820** [comment]
closing as not planned
